src/sys/socket/multicast.rs

```diff
@@ -37,4 +37,10 @@ impl ipv6_mreq {
             ipv6mr_interface: 0,
         }
     }
+    pub fn new(group: Ipv6Addr, interface: c_uint) -> ipv6_mreq {
```

Comment:
This (function overloading based on number of arguments) isn't supported in rust AFAIK -- you can do it based on type by using traits but I don't think that works in this case.  We'll either have to break backwards compatibility or introduce a new function name.  I think the  latter probably makes sense.